tests/unit/test__main.py

```diff
@@ -162,3 +172,35 @@ def test_config_runner_3(mock_args, mock_build, mock_run, caplog):
     assert mock_run.call_args.args[0].config_dir is None
     assert os.path.basename(mock_run.call_args.args[0].config_file) == "0002.yaml"
     assert len(mock_run.call_args.args[1]) == 1
+
+
+@mock.patch("data_validation.__main__.run_validations")
+@mock.patch(
+    "data_validation.__main__.build_config_managers_from_yaml",
+    return_value=["config dict from one file"],
+)
+@mock.patch(
+    "argparse.ArgumentParser.parse_args",
+    return_value=argparse.Namespace(**CONFIG_RUNNER_ARGS_4),
+)
+def test_config_runner_4(mock_args, mock_build, mock_run, caplog):
+    """Third test - run validation on a directory with failures in one validation,
+        Running in a non Kube completions environment. Expected Result:
+    1. All 4 files are validated, even though one of them raises an exception.
+    2. Exception from one validation is trapped, file skipped and raised at the end.
+    """
+    mock_run.side_effect = [10, ValueError("Boom!"), 12, 10]
+    caplog.set_level(logging.WARNING)
+    args = cli_tools.get_parsed_args()
+    caplog.clear()
+    with pytest.raises(Exception) as e_info:
+        main.config_runner(args)
+    # assert that exception message was output for the failed validation
+    # validation is called four times, once for each file
+    # After all four files were validated, an exception was raised back to main to return status
+    assert (
+        caplog.messages[0]
+        == "Error Boom! occurred while running config file 0001.yaml. Skipping it for now."
```

Comment:
Seeing as we are using this text as an assertion for a test I wonder if we should convert it to a constant?

For example:
```
assert (
        caplog.messages[0]
        == CONFIG_RUNNER_EXCEPTION_TEXT.format("Boom!, "0001.yaml")
)
```

Then if the message text changes the assertion automatically follows suit.